allennlp/data/semparse/worlds/nlvr_world.py

```diff
@@ -126,6 +126,61 @@ def _apply_function_list(self,
             return_value = getattr(self, function_name)(return_value)
         return return_value
 
+    def get_logical_form(self,
+                         action_sequence: List[str],
+                         add_var_function: bool = True) -> str:
+        """
+        Takes an action sequence and constructs a logical form from it. This is useful if you want
+        to get a logical form from a decoded sequence of actions generated by a transition based
+        semantic parser.
+
+        Parameters
+        ----------
+        action_sequence : ``List[str]``
+            The sequence of actions as strings (eg.: ``['@START@ -> t', 't -> <e,t>', ...]``).
+        add_var_function : ``bool`` (optional)
+             ``var`` is a special function that some languages use within lambda functions to
+             indicate the use of a variable (eg.: ``(lambda x (fb:row.row.year (var x)))``). Due to
+             the way constrained decoding is currently implemented, it is easier for the decoder to
+             not produce these functions. In that case, setting this flag adds the function in the
+             logical form even though it is not present in the action sequence.
+        """
+        # TODO(pradeep): This method almost works with LambdaDCS logical forms in WikiTables as
+        # well. The thing that breaks it is "reverse", which is a higher order function. Once I
```

Comment:
Yes, since `reverse` is a higher order function. `infer_num_arguments` will return `2` for `reverse`, whereas it takes only one argument (the function being reversed), and returns another function, and not a basic type like the other functions.